### PR TITLE
Do not collapse editor lines in single editor

### PR DESCRIPTION
**Pull request by fcollonval**
- Don't collapsed lines in editor for single pane merge view.
- Correctly bind label and inputs

Fix partly https://github.com/jupyter/nbdime/issues/718 (it does not solve the notebook metadata unchanged case).

### CODE REVIEW

**Comment by fcollonval** [comment]
bot please update playwright snapshots

File: packages/nbdime/src/common/mergeview.ts [hunk 1]
```diff
@@ -1444,7 +1444,13 @@ export class MergeView extends Panel {
     const additionalExtensions = inMergeView
       ? [listener, mergeControlGutter, getCommonEditorExtensions(inMergeView)]
       : getCommonEditorExtensions(inMergeView);
-    if (this._collapseIdentical >= 0) {
+    const singlePane = !merged && (remote?.unchanged || remote?.added || remote?.deleted)
+    if (
+      // no collapse
+      this._collapseIdentical >= 0 &&
+      // display a single editor
```
Comment (fcollonval) [inline packages/nbdime/src/common/mergeview.ts:1451]:
```suggestion
      // not displaying a single editor
```

File: nbdime/webapp/templates/compare.html [hunk 2]
```diff
@@ -9,18 +9,18 @@ <h3>Notebook Diff/Merge</h3>
       <form id="nbdime-compare-form" class="nbdime-forms">
         <fieldset>
           <legend>Please input filenames of notebooks to compare:</legend>
-          <label>Local:</label>
-          <input id="compare-local" type="text" name="local" value="{{ config_data['local']|e }}" />
-          <label>Base:</label>
-          <input id="compare-base" type="text" name="base" value="{{ config_data['base']|e }}" />
-          <label>Remote:</label>
-          <input id="compare-remote" type="text" name="remote" value="{{ config_data['remote']|e }}" />
+          <label>Local:
```
Comment (krassowski) [inline nbdime/webapp/templates/compare.html:12]: Keeping atomic labels with `for="id"`  might make it easier to translate in the #709 but for the sake of this one it does not matter much.

File: packages/nbdime/src/common/mergeview.ts [hunk 3]
```diff
@@ -1444,7 +1444,13 @@ export class MergeView extends Panel {
     const additionalExtensions = inMergeView
       ? [listener, mergeControlGutter, getCommonEditorExtensions(inMergeView)]
       : getCommonEditorExtensions(inMergeView);
-    if (this._collapseIdentical >= 0) {
+    const singlePane = !merged && (remote?.unchanged || remote?.added || remote?.deleted)
+    if (
+      // no collapse
+      this._collapseIdentical >= 0 &&
+      // not displaying a single editor
+      !singlePane
```
Comment (krassowski) [inline packages/nbdime/src/common/mergeview.ts:1452]: Note: tested locally to confirm that it also works for diff view.

**Review by krassowski — APPROVED** [review]
Tested locally, confirmed that it fixes the issue and other changes are an improvement too although the same bits may need changes for translation work later.

**Pull request merged**